x/tier/keeper/portfolio.go

```diff
@@ -271,6 +271,42 @@ func (k Keeper) RetrieveLeverageLpTotal(ctx sdk.Context, user sdk.AccAddress) sd
 	return totalValue
 }
 
+func (k Keeper) RetrieveLeverageLpTotalAssets(ctx sdk.Context, user sdk.AccAddress) sdk.Dec {
```

Comment:
I don't think we need copy of same logic in 3 functions. We can reuse `RetrieveLeverageLpTotal` function